tools/test-framework/src/relayer_mock/base/types/state.rs

```diff
@@ -0,0 +1,47 @@
+use std::collections::HashSet;
+use std::fmt::Display;
+
+#[derive(Clone, Default)]
+pub struct State {
+    recv_events: HashSet<String>,
```

Comment:
We can use `HashSet<(PortId, ChannelId, Sequence)>` as key.

It would be good to define type aliases such as `type PortId = String`, so that we can change it to other types like `u8` easily in the future.